pkg/azurediskplugin/main.go

```diff
@@ -80,44 +97,138 @@ func main() {
+		}
+
+		// Initialize driveConfig object with default values
+		driverConfig = azdiskv1beta2.AzDiskDriverConfiguration{
+			ControllerConfig: azdiskv1beta2.ControllerConfiguration{
+				DisableAVSetNodes:       consts.DefaultDisableAVSetNodes,
+				VMType:                  consts.DefaultVMType,
+				EnableDiskOnlineResize:  consts.DefaultEnableDiskOnlineResize,
+				EnableAsyncAttach:       consts.DefaultEnableAsyncAttach,
+				EnableListVolumes:       consts.DefaultEnableListVolumes,
+				EnableListSnapshots:     consts.DefaultEnableListSnapshots,
+				EnableDiskCapacityCheck: consts.DefaultEnableDiskCapacityCheck,
+				Enabled:                 consts.DefaultIsControllerPlugin,
+				LeaseDurationInSec:      consts.DefaultControllerLeaseDurationInSec,
+				LeaseRenewDeadlineInSec: consts.DefaultControllerLeaseRenewDeadlineInSec,
+				LeaseRetryPeriodInSec:   consts.DefaultControllerLeaseRetryPeriodInSec,
+				LeaderElectionNamespace: consts.ReleaseNamespace,
+				PartitionName:           consts.DefaultControllerPartitionName,
+			},
+			NodeConfig: azdiskv1beta2.NodeConfiguration{
+				VolumeAttachLimit:       consts.DefaultVolumeAttachLimit,
+				SupportZone:             consts.DefaultSupportZone,
+				EnablePerfOptimization:  consts.DefaultEnablePerfOptimization,
+				UseCSIProxyGAInterface:  consts.DefaultUseCSIProxyGAInterface,
+				GetNodeInfoFromLabels:   consts.DefaultGetNodeInfoFromLabels,
+				Enabled:                 consts.DefaultIsNodePlugin,
+				HeartbeatFrequencyInSec: consts.DefaultHeartbeatFrequencyInSec,
+				PartitionName:           consts.DefaultNodePartitionName,
+			},
+			CloudConfig: azdiskv1beta2.CloudConfiguration{
+				SecretName:            consts.DefaultCloudConfigSecretName,
+				SecretNamespace:       consts.DefaultCloudConfigSecretNamespace,
+				CustomUserAgent:       consts.DefaultCustomUserAgent,
+				UserAgentSuffix:       consts.DefaultUserAgentSuffix,
+				AllowEmptyCloudConfig: consts.DefaultAllowEmptyCloudConfig,
+				VMSSCacheTTLInSeconds: consts.DefaultVMSSCacheTTLInSeconds,
+			},
+			ClientConfig: azdiskv1beta2.ClientConfiguration{
+				Kubeconfig:    consts.DefaultKubeconfig,
+				KubeClientQPS: consts.DefaultKubeClientQPS,
+			},
+			ObjectNamespace: consts.DefaultAzureDiskCrdNamespace,
+			Endpoint:        consts.DefaultEndpoint,
+			MetricsAddress:  consts.DefaultMetricsAddress,
+			DriverName:      consts.DefaultDriverName,
+		}
+
+		// Convert yaml to a driveConfig object
+		err = yaml.Unmarshal(yamlFile, &driverConfig)
+		if err != nil {
+			klog.Fatalf("failed to unmarshal the driver config, error: %v", err)
+		}
+
+		// Emit warning log for using deprecated command-line parameters
+		flag.Visit(func(f *flag.Flag) {
+			if slices.Contains(consts.CommandLineParams, f.Name) {
+				klog.Warningf("the command-line parameter %v is deprecated and overridden by --config parameter.", f.Name)
+			}
+		})
+	} else {
+		driverConfig.Endpoint = *endpoint
```

Comment:
np: for consistency, lets use the same style as above (lines 115-157) when initializing the parameters.